### DISCUSSION TITLE
how does trino compaction work

### DISCUSSION THREAD

**Discussion by JunseoChoJJ** [trinodb/trino#22453] [Q&A]
@findepi 
I had 371 parquet file it's about 500kb each and total 240mb
and I compact using trino statement

`alter table nessie_catalog.nessie.kafka2 execute optimize(file_size_threshold => '128MB')`

and I got 3 parquet file. 85mb, 60mb, 45mb 
I wonder if I use that statement above I would get 128mb, 60mb just two parquet file

how does trino compaction work?

**Comment by findepi** [comment]
Exact file size is not guaranteed, as Trino does not aim to create **minimal** number of files.
This is also subject to writer scaling.
